### DISCUSSION TITLE
Livewire Form Not Saving 1 Column

### DISCUSSION THREAD

**Discussion by drdan18** [livewire/livewire#4235] [1. Help] [answered]
I have a livewire component with a form. 

The data from the form is being validated and the returned data looks good.  

There is one field within that data like this "profession_id" => "1".   It is a BigInteger column so this looks good.  

There are 7 other fields being saved along with it.  Just this column does not get saved.  No errors are giving by the database.   The field is nullable and that what it gets set to. 

I hope hoping someone else has seen a similar situation.  With a simple solution of course. 

Thanks. 

Dan

**Comment by joshhanley** [comment]
@drdan18 hard to say without seeing the code. Can you share your component class?

Thanks!

**Reply by drdan18** [reply]
Thanks for your reply. 

I figured it out.   The problem was the column was missing from my $fillable array in the Model. 

I would expect a mass assignment error in this situation.  Is that something that livewire ignores?

**Reply by joshhanley** [reply]
@drdan18 ah interesting, I'm not sure, if Laravel throws an error, I'd expect Livewire to just forward that error to the browser.

**Reply by joshhanley** [reply]
I never use `$fillable` so don't know what happens 😅

**Reply by drdan18** [reply]
LOL.  Fair enough.  I just started using it but might reconsider now.

**Reply by joshhanley** [reply]
Haha yeah, always just use `$guarded = [];`.

I think Jeffery Way has a good video on why to not use fillable, but the TLDR is imagine you have a flag like `is_admin`, well users shouldn't be able to change that, so you make sure it isn't in `fillable`.

But then you go to build an admin section, who can manage the `is_admin` property, and you then need add it to the fillable array. You then potentially have a security concern in the users form if they then send `is_admin` along with the request.

Better instead to not use the protection at the model level and use `$guarded = [];`, then make sure you do your checks at the controller level using validation, than then ensures only validate data for that specific request get passed onto the model.

Hope that helps!

**Reply by drdan18** [reply]
Thanks for the info.  That does help. 

I knew Jeffery just uses the $guarded = [ ];  but I never knew the reasoning.  Thanks for explaining and that makes total sense now.

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by drdan18** [comment, marked as answer]
See above
